### DISCUSSION TITLE
ERR Log "attempted to connect to route port"

### DISCUSSION THREAD

**Discussion by hufan-ola** [nats-io/nats-server#3569] [Q&A] [answered]
In Cluster model, when I pub a message to cluster addr, subscribe **can receive** message, but  we got an ERR log
```
nats sub foo -s nats://192.168.0.3:6222

[15355] 2022/10/18 11:30:00.638646 [ERR] 192.168.0.3:57824 - rid:10 - attempted to connect to route port
```
what’s the problem, what should i do ?
Please

config file as below
- main.conf
```
listen: 192.168.0.3:4222
http: 192.168.0.3:8222
debug: true
trace: false

cluster {
  name: pressure1
  listen: 192.168.0.3:6222
}

```

- slave.conf
```
listen: 192.168.0.3:4223
http: 192.168.0.3:8223
debug: true
trace: false

cluster {
  name: pressure1
  listen: 192.168.0.3:6223

  routes = [
    nats-route://192.168.0.3:6222
  ]
}
```

**Comment by Jarema** [comment, marked as answer]
Hey!

`cluster` routes are meant to be used only for other nodes (servers) to connect and form a cluster.

Clients should connect to the listen URL of one of the nodes. In your example it's `192.168.0.3:4222`.

Docs are here: https://docs.nats.io/running-a-nats-service/configuration/clustering

Keep in mind that your cluster should have at least 3 nodes.
If you want master/slave hub/spoke architecture, [leafnodes](https://docs.nats.io/running-a-nats-service/configuration/leafnodes) might be a better solution.

**Reply by hufan-ola** [reply]
<img width="527" alt="image" src="https://user-images.githubusercontent.com/113106990/196359819-d17490f5-0d13-445b-a791-0597d39173ee.png">

SubA listen for msg
```
nats -s nats://192.168.0.243:4222 sub foo
```

SubA can received msg below three port for publish 
```
nats -s nats://192.168.0.243:4222 pub foo h1
nats -s nats://192.168.0.243:4223 pub foo h2
nats -s nats://192.168.0.243:4224 pub foo h3
```

But if Seed Node crashed, SubA can attempt reconnect to port 4223 automatic
```
Disconnected due to: EOF, will attempt reconnect
Reconnected [nats://192.168.0.243:4223]
```

but client will report an error as below
```
nats -s nats://192.168.0.243:4222 pub foo h1
nats: error: nats: no servers available for connection
```

What I need is client can use **one address** to connect the Cluster, and do not need appoint specific address(eg: 192.168.0.243:4222) and client can publish **success** as long as any node in Cluster is alive

Can we did it?   @Jarema

**Reply by Jarema** [reply]
When connecting to any of the nodes, Client will get information on other nodes in the cluster. That's why your `subscribe` example works by automatically reconnecting.

If you need to be able to **connect** to the cluster, no matter which servers are up, using only one URL, that's out of the scope of servers setup. 
You probably need something like a DNS with multiple entries, or [something relevant to your setup](https://docs.nats.io/running-a-nats-service/nats-kubernetes/nats-external-nlb#using-a-load-balancer-for-external-access-to-nats). 

You can also pass more URLs to the connect `nats pub foo data  --server localhost:4222,localhost:4223` and client will connect to the one that is available.
